### ISSUE TITLE
Missing comma in `-doto`

### ISSUE THREAD

**Issue by SwiftLawnGnome** [magnars/dash.el#333]
When a form in the body of `-doto` is not a list, the form isn't unquoted into the `funcall`, so
```elisp
(-doto (make-sparse-keymap)
  use-local-map)
```
Becomes 
```emacs-lisp
(let ((#:value (make-sparse-keymap)))
  (funcall form #:value)
  #:value)
```
Also, why is a `funcall` used in this case, as opposed to threading a la `->`? That means the use of a symbol in the body is only useful when calling a function that's the value of a variable, which seems like a pretty uncommon case.

**Comment by Fuco1** [comment]
I'm not sure I understand.  Can you give an example?  To be quite honest I never used `-doto` and I don't fully understand how it is useful.

**Comment by SwiftLawnGnome** [comment]
Sure
```elisp
(macroexpand '(-doto "hello" message))
```
returns
```elisp
(let ((#:value "hello"))
  (funcall form #:value)
  #:value)
```
In the expansion, the actual symbol `form` (instead of `message`) is in the expansion, because the macro uses this lambda:
```elisp
(lambda (form)
  (if (sequencep form)
      `(,(-first-item form) ,retval ,@(cdr form))
    `(funcall form ,retval)))
```
As you can see on the last line, `form` isn't unquoted.

To my second point, if the macro was fixed, it expands to
```elisp
(let ((#:value "hello"))
  (funcall message #:value)
  #:value)
```
this would error unless the symbol named `message` was bound by value to a function. I personally would rewrite the macro to be
```elisp
(defmacro -doto (eval-initial-value &rest forms)
  (let ((retval (make-symbol "value")))
    `(let ((,retval ,eval-initial-value))
       ,@(--map `(-> ,retval ,it) forms)
       ,retval)))
```
So the threading behavior is the same as `->`, meaning the `symbol-function` of `message` would be called.

**Comment by basil-conto** [comment]
I just ran into this issue myself. :) Thanks for the suggestion! Should be fixed in commit d983d4fea036b422149e13c473ecead2f7b8e4a8.
